module/VuFind/src/VuFind/Controller/AbstractSearch.php

```diff
@@ -389,6 +385,11 @@ protected function getSearchResultsView($setupCallback = null)
                 $this->saveSearchToHistory($results);
             }
 
+            // If a "jumpto" parameter is set, deal with that now:
+            if ($jump = $this->processJumpTo($results)) {
```

Comment:
I thought if we were jumping to a specific record, we wouldn't want to remember a search in the search history. I imagine you had to refactor this to ensure that the sid was available to the processJumpTo() method, but does this have undesirable side effects? Might there be a better way to do this?

Reply:
I find it odd that a search is not saved if it finds only a single record and jump_to_single_search_result is enabled, but it is if jump_to_single_search_result is disabled. Currently the correct search tab is lost in the former case, and that's why I changed it, but it could be changed back if that's more desirable.
 
In other cases I actually don't quite understand how jumpto works and why it's needed. :)

Reply:
Ahh, I did some searching and found the reason for it -- it's for the MARC record linker, which uses jumpto links to link to records that it searches via identifiers. See https://github.com/vufind-org/vufind/blob/61f273c0a2db5f13b840821023ea34740b21522a/module/VuFind/src/VuFind/View/Helper/Root/RecordLinker.php#L75 for an example. I definitely think we do NOT want these links to add entries to search history.

Reply:
I revised it so that jumpto doesn't get added to history, and single result jump is handled separately.